cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSetArguments.java

```diff
@@ -126,34 +114,23 @@ public static void setInputPartitionFilter(Map<String, String> arguments, Partit
    * @param arguments the runtime arguments for a partitioned dataset
    * @param partitioning the declared partitioning for the dataset, needed for proper interpretation of values
    * @return the PartitionFilter specified in the arguments or null if no filter is specified.
+   * @deprecated as of 3.4.0. Use {@link #getInputPartitionFilter(Map)} instead.
    */
   @Nullable
+  @Deprecated
   public static PartitionFilter getInputPartitionFilter(Map<String, String> arguments, Partitioning partitioning) {
-    PartitionFilter.Builder builder = PartitionFilter.builder();
-    for (Map.Entry<String, FieldType> entry : partitioning.getFields().entrySet()) {
-      String fieldName = entry.getKey();
-      FieldType fieldType = entry.getValue();
+    return getInputPartitionFilter(arguments);
+  }
 
-      // is it a single-value condition?
-      String stringValue = arguments.get(INPUT_PARTITION_VALUE_PREFIX + fieldName);
-      Comparable fieldValue = convertFieldValue("filter", "value", fieldName, fieldType, stringValue, true);
-      if (null != fieldValue) {
-        @SuppressWarnings({ "unchecked", "unused" }) // we know it's type safe, but Java does not
-        PartitionFilter.Builder unused = builder.addValueCondition(fieldName, fieldValue);
-        continue;
-      }
-      // must be a range condition
-      String stringLower = arguments.get(INPUT_PARTITION_LOWER_PREFIX + fieldName);
-      String stringUpper = arguments.get(INPUT_PARTITION_UPPER_PREFIX + fieldName);
-      Comparable lowerValue = convertFieldValue("filter", "lower bound", fieldName, fieldType, stringLower, true);
-      Comparable upperValue = convertFieldValue("filter", "upper bound", fieldName, fieldType, stringUpper, true);
-      if (null == lowerValue && null == upperValue) {
-        continue; // this field was not present in the filter
-      }
-      @SuppressWarnings({ "unchecked", "unused" }) // we know it's type safe, but Java does not
-      PartitionFilter.Builder unused = builder.addRangeCondition(fieldName, lowerValue, upperValue);
-    }
-    return builder.isEmpty() ? null : builder.build();
+  /**
+   * Get the partition filter for the input to be read.
+   *
+   * @param arguments the runtime arguments for a partitioned dataset
+   * @return the PartitionFilter specified in the arguments or null if no filter is specified.
+   */
+  @Nullable
+  public static PartitionFilter getInputPartitionFilter(Map<String, String> arguments) {
+    return GSON.fromJson(arguments.get(INPUT_PARTITION_FILTER), PartitionFilter.class);
```

Comment:
should we support the old style for backwards compatibility? If someone is passing in runtime arguments from outside instead of creating them programmatically?

In that case, accept the old form and log a warning that this will be discontinued in the future?